test/nn/encoder/test_stype_encoder.py

```diff
@@ -426,3 +432,148 @@ def test_text_tokenized_encoder():
+                                     col_names)
+        assert x.shape == (
+            num_rows,
+            len(tensor_frame.col_names_dict[data_stype]),
+            out_channels,
+        )
+
+
+class EmbeddingModel(torch.nn.Module):
+    def __init__(self, in_channels: int, out_channels: int):
+        super().__init__()
+        self.mlp = Sequential(Linear(in_channels, out_channels), ReLU(),
+                              Linear(out_channels, out_channels))
+
+    def forward(self, x: MultiEmbeddingTensor) -> Tensor:
+        # [batch_size, 1, embedding_size]
+        return self.mlp(x.values.unsqueeze(dim=1))
+
+
+class NumericalModel(torch.nn.Module):
+    def __init__(self, in_channels: int, out_channels: int):
+        super().__init__()
+        self.mlp = Sequential(Linear(in_channels, out_channels), ReLU(),
+                              Linear(out_channels, out_channels))
+
+    def forward(self, x: Tensor) -> Tensor:
+        # [batch_size, 1, 1] -> [batch_size, 1, out_channels]
+        return self.mlp(x)
+
+
+class TimestampModel(torch.nn.Module):
+    def __init__(self, in_channels: int, out_channels: int):
+        super().__init__()
+        self.weight = torch.nn.Parameter(
+            torch.empty(in_channels, out_channels, out_channels))
+        self.cyclic_encoding = CyclicEncoding(out_size=out_channels)
+
+    def forward(self, x: Tensor) -> Tensor:
+        # [batch_size, 1, num_time_feats]
+        x = x.to(torch.float32)
+        # [batch_size, 1, num_time_feats, out_channels]
+        x_cyclic = self.cyclic_encoding(x / x.max())
+        # [batch_size, 1, out_channels]
+        return torch.einsum('ijk,jkl->il', x_cyclic.squeeze(1),
+                            self.weight).unsqueeze(dim=1)
+
+
+class CategoricalModel(torch.nn.Module):
+    def __init__(self, num_categories: int, out_channels: int):
+        super().__init__()
+        self.emb = torch.nn.Embedding(num_categories, out_channels)
+
+    def forward(self, x: Tensor) -> Tensor:
+        # [batch_size, 1, 1] -> [batch_size, 1]
+        x = x.squeeze(dim=1)
+        # [batch_size, 1] -> [batch_size, 1, out_channels]
+        return self.emb(x)
+
+
+class MultiCategoricalModel(torch.nn.Module):
+    def __init__(self, num_categories: int, out_channels: int):
+        super().__init__()
+        self.emb = torch.nn.EmbeddingBag(num_categories, out_channels)
+
+    def forward(self, x: MultiNestedTensor) -> Tensor:
+        return self.emb(x.values, x.offset[:-1]).unsqueeze(dim=1)
+
+
+class SequenceNumericalModel(torch.nn.Module):
+    def __init__(self, in_channels: int, out_channels: int):
+        super().__init__()
+        self.mlp = Sequential(Linear(in_channels, out_channels), ReLU(),
+                              Linear(out_channels, out_channels))
+
+    def forward(self, x: MultiNestedTensor) -> Tensor:
+        # [batch_size, 1, max_length]
+        return self.mlp(x.to_dense(fill_value=0.0))
```

Comment:
Weird to apply MLP here. It should be sequence model like RNN, but I don't think we need to support sequence numerical here.